app/graphql/types/query_type.rb

```diff
@@ -0,0 +1,128 @@
+module Types
+  class QueryType < Types::BaseObject
+    field :comments, Types::CommentType.connection_type, "Returns all comments", null: false
+    field :comment, Types::CommentType, "Returns comment for ID", null: false do
+      argument :id, ID, required: true, default_value: false
+    end
+
+    field :debates, Types::DebateType.connection_type, "Returns all debates", null: false
+    field :debate, Types::DebateType, "Returns debate for ID", null: false do
+      argument :id, ID, required: true, default_value: false
+    end
+
+    field :geozones, Types::GeozoneType.connection_type, "Returns all geozones", null: false
+    field :geozone, Types::GeozoneType, "Returns geozone for ID", null: false do
+      argument :id, ID, required: true, default_value: false
+    end
+
+    field :proposals, Types::ProposalType.connection_type, "Returns all proposals", null: false
+    field :proposal, Types::ProposalType, "Returns proposal for ID", null: false do
+      argument :id, ID, required: true, default_value: false
+    end
+
+    field :proposal_notifications, Types::ProposalNotificationType.connection_type, "Returns all proposal notifications", null: false
+    field :proposal_notification, Types::ProposalNotificationType, "Returns proposal notification for ID", null: false do
+      argument :id, ID, required: true, default_value: false
+    end
+
+    # Support deprecated API
+    field(
+      :proposal_notifications,
+      Types::ProposalNotificationType.connection_type,
+      "Returns all proposal notifications",
+      null: false,
+      camelize: false,
+      deprecation_reason: "Snake case fields are deprecated. Please use proposalNotifications."
+    )
+
+    field(
+      :proposal_notification,
+      Types::ProposalNotificationType,
+      "Returns proposal notification for ID",
+      null: false,
+      camelize: false,
+      deprecation_reason: "Snake case fields are deprecated. Please use proposalNotification."
+    ) do
+      argument :id, ID, required: true, default_value: false
+    end
+
+    field :tags, Types::TagType.connection_type, "Returns all tags", null: false
+    field :tag, Types::TagType, "Returns tag for ID", null: false do
+      argument :id, ID, required: true, default_value: false
+    end
+
+    field :users, Types::UserType.connection_type, "Returns all users", null: false
+    field :user, Types::UserType, "Returns user for ID", null: false do
+      argument :id, ID, required: true, default_value: false
+    end
+
+    field :votes, Types::VoteType.connection_type, "Returns all votes", null: false
+    field :vote, Types::VoteType, "Returns vote for ID", null: false do
+      argument :id, ID, required: true, default_value: false
+    end
+
+    def comments
+      Comment.public_for_api
+    end
+
+    def comment(id:)
+      Comment.find(id)
```

Comment:
Originally, we had this line to find a record:

```ruby
model.public_for_api.find_by(id: ...)
```

There are two differences here:

1. (Most important) We used the `public_for_api` scope to guarantee only public records could be accessed
2. We returned `nil` instead of raising an exception

I guess we can change this line:

```suggestion
      Comment.public_for_api.find_by(id)
```

And replace `null: false` with `null: true` in the `field` method :thinking:.

Since we have to make this change for every model, maybe we could refactor the code a little bit:

```ruby
    {
      comment: Types::CommentType,
      debate: Types::DebateType,
      geozone: Types::GeozoneType,
      proposal: Types::ProposalType,
      proposal_notification: Types::ProposalNotificationType,
      tag: Types::TagType,
      user: Types::UserType,
      vote: Types::VoteType
    }.each do |field_name, type_class|
      plural_name = field_name.to_s.pluralize
      model_class = field_name.to_s.classify.constantize

      field plural_name, type_class.connection_type, "Returns all #{plural_name.gsub("_", " ")}", null: false
      field field_name, type_class, "Returns #{field_name} for a given ID", null: true do
        argument :id, ID, required: true, default_value: false
      end

      define_method(field_name) { |id:| model_class.public_for_api.find_by(id: id) }
      define_method(plural_name) { model_class.public_for_api }
    end
```

Not sure whether this is an improvement, though :thinking:. I'll leave it up to you whether to apply this refactoring :wink:.

Reply:
Actually, if you would have asked me a few years ago, I would probably said: This is a really good idea.
But recent projects have taught me that very often, code that is a bit more verbose, but 'standard' according to the doc of whatever library you are using, is a better choice.

This change would re-introduce some of the complexity that I removed with the 'weird' `ApiTypesCreator`, so I think I wouldn't want to do it.
Also: You never know what's going to happen in the future. You might end up with edge cases for some models and than you would have to introduce more complexity here.

Reply:
Yes, that's why I wasn't sure about it. As mentioned, up to you! :wink: